packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -56,6 +56,91 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
   onSelect?.(selectedData, newState);
 };
 
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance: TableInstance<T>,
+) => {
+  const resetSelection = () => {
+    instance.rows.forEach((r) => r.toggleRowSelected(false));
+  };
+
+  const getGreaterId = (id: string) => {
+    const greaterId = id.split('.');
+    greaterId[greaterId.length - 1] = `${
+      parseInt(greaterId[greaterId.length - 1]) + 1
+    }`;
+    return greaterId.join('.');
+  };
+
+  const getParentId = (id: string) => {
+    const lastDotIndex = id.lastIndexOf('.');
+    if (lastDotIndex < 0) {
+      return null;
+    }
+    return id.substring(0, lastDotIndex);
+  };
+
+  // Returns true if endId is a parent of currentId
+  // E.g. currentId = 1.4.2.3, endId = 1.4  -> true
+  // E.g. currentId = 1.2.5, endId = 1.3    -> false
+  const isAncestorParent = (currentId: string, endId: string) => {
+    return endId.indexOf(currentId) === 0;
+  };
+
+  // Returns the next row as if it was just a non subrows list
+  const getNextRow = (currentRow: Row<T>, startId: string, endId: string) => {
```

Comment:
`startId` param is declared but never read in current approach.

Reply:
**[DONE]** Ohh, thanks, `startId` is removed now